server/server.ts

```diff
@@ -18,131 +19,108 @@ var user_id = 0;
-
-      socket.emit("gameplay", user1 );
-      socket.to(user1.socketId).emit("gameplay", user2 ); // sends message to specific client based on clientId
-      current_state = 0;
-
-    // for excess users, cya 
+  socket.on('hello', (data: string) => {
+    let msg: ClientSentUserInfo = JSON.parse(data);
+    console.log(msg);
+    if (!msg.isInvalid) {
+      // if user is first user, setup
+      if (user1 === undefined || user1.id === msg.id) {
+        user1 = new UserInfo(msg.name, msg.id, socket.id, 1, Math.random());
+        // Returns 1 to indicate player 1
+        socket.emit("pairing", 1);
+
+        // if user is second user, setup and allow for starting game by telling clients
+      } else if (user2 === undefined || user2.id === msg.id) {
+        user2 = new UserInfo(msg.name, msg.id, socket.id, 2, user1.randomSeed);
+        // Returns 2 to indicate player 2
+        socket.emit("pairing", 2);
+        socket.to(user1.socketId).emit("gameplay", user2.exportClientRequiredUserInfo()); // sends message to specific client based on clientId
+        socket.emit("gameplay", user1.exportClientRequiredUserInfo());
+
+        // for excess users, cya 
+      } else {
+        socket.emit("pairing", 'There are already 2 players. Please try again later.');
+      }
     } else {
-      socket.emit("initiate", 'There are already 2 players. Please try again later.');
+      socket.emit("invalid", "invalid user data")
     }
   });
 
   /**
   * Receives turn info from users, and sends turn info to other player once both
   *   players have submitted
-  * Recall that for current_state:
-  *   0 : no commands submitted, 1: P1 only submitted, 2: P2 only submitted, 3: both submitted
+  * TODO: CHECK IF GAME STARTED IN MULTIGAME
   */
-  socket.on('submittingTurn', (msg) => {
-    //console.log("Server has received commands from a user");
-    msg = JSON.parse(msg);
-
-    // Check which user has submitted and change state accordingly
-    if(user1.id == msg.playerId){
-      user1.commands = msg.commandString;
-      if(current_state == 0 || current_state == 2) {
-        current_state++;
-      }  
-      console.log("Received commands from player 1");
-      console.log("Current State: " + current_state);
-    }
-    else if(user2.id == msg.playerId){
-      user2.commands = msg.commandString;
-      if(current_state == 0 || current_state == 1) {
-        current_state += 2;
-      }  
-      console.log("Received commands from player 2");
-      console.log("Current State: " + current_state);
-    }
-
-    // If we've received both sets of commands, send back to players
-    if(current_state == 3) {
-      socket.to(user1.socketId).emit("receiveTurn", user2.commands);
-      socket.to(user2.socketId).emit("receiveTurn", user1.commands);
-      console.log("Sending commands to both players")
-      current_state = 0;
+  socket.on('submittingTurn', (data: string) => {
+    let msg: SocketTurnInfo = JSON.parse(data);
+    if (!msg.isInvalid) {
```

Comment:
change to checking if commands are null in order to send. this is because for multigame, its much easier to store a singular object (UserInfo) in a map of UserInfos or something liek that